### PR TITLE
fix: Dbtp-1094 use platform-helper notify for Slack alert on publish to PyPi

### PR DESCRIPTION
**Pull request by tony-griffin**
Addresses [DBTP-1094](https://uktrade.atlassian.net/browse/DBTP-1094)

- Previous logic to send message to Slack on publish to PyPi used the Slack WebClient directly. This failed in the codebuild job when the logic tried to use the WebClient as it could not be found.
- Now we are using the inbuilt `platform-helper notify` command to alert when a new version of platform-tools is published as this is becoming the standard way we alert to Slack.
- Tests updated
- Format of message: 
Screenshot 2024-07-05 at 15 01 22

---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- [x] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [x] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing - [platform-tools-test codebuild execution](https://eu-west-2.console.aws.amazon.com/codesuite/codebuild/763451185160/projects/platform-tools-test/build/platform-tools-test%3A1f7197d4-9311-488c-949d-0514cd98f2bc?region=eu-west-2)

[DBTP-1094]: https://uktrade.atlassian.net/browse/DBTP-1094?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: buildspec.pypi.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ env:
     PYPI_TOKEN: "pypi-token"
     SLACK_TOKEN: "/codebuild/slack_oauth_token"
     SLACK_CHANNEL_ID: "/codebuild/slack_platform_helper_publish_oauth_channel"
+
```
Comment (gabelton) [inline buildspec.pypi.yml:9]:
```suggestion
```

File: tests/utils/test_publish_notifications.py [hunk 2]
```diff
@@ -90,56 +104,14 @@ def test_version_provided_must_be_a_string(self, webclient, time):
...
-                    text=f"*Version*: {version}",
-                ),
-                blocks.TextObject(
-                    type="mrkdwn",
-                    text=f"<{RELEASE_NOTES_URL_LATEST}|Release Notes>",
-                ),
-            ]
-        ),
-    ]
+if __name__ == "__main__":
```
Comment (gabelton) [inline tests/utils/test_publish_notifications.py:116]: Why is this needed?

File: buildspec.pypi.yml [hunk 3]
```diff
@@ -18,8 +19,10 @@ phases:
       - echo -e "\nInstalling dependencies"
       - pip install poetry
       - poetry install
+      - pip install dbt-platform-helper
 
   build:
     commands:
       - echo -e "\nBuild and publish to PyPi"
       - ./publish_to_pypi.sh ${PYPI_TOKEN} ${SLACK_TOKEN} ${SLACK_CHANNEL_ID}
+
```
Comment (gabelton) [inline buildspec.pypi.yml:28]: github complaining about lack of newline at EOF, despite newline...

File: tests/utils/test_publish_notifications.py (hunk 2 above)
Comment (tony-griffin) [reply tests/utils/test_publish_notifications.py]: Will remove 👍🏻

File: buildspec.pypi.yml (hunk 3 above)
Comment (tony-griffin) [reply buildspec.pypi.yml]: @gabelton Have removed newline to follow pattern of other buildspec files

**Review by ksugden — APPROVED** [review]
suggestion to change the merge commit to describe the change rather than the problem:
fix: DBTP-1094 - platform tools publishes 'pypi package published' slack notification
